test/utils.py

```diff
@@ -214,3 +215,63 @@ def test_should_get_all_without_params(self):
         self._testing_class().run()
 
         self._resource_client.get_all.assert_called_once_with()
+
+
+class ErrorHandlingTestCase(PreloadedMocksBaseTestCase):
+    """
+    ErrorHandlingTestCase has common test for the modules error handling.
+    """
+    params_present = dict(
+        config='config.json',
+        state='present',
+        data=dict(name='Resource Identifier'))
+
+    error_message = 'Fake message error'
+
+    def configure_client_mock(self, resorce_client):
+        """
+        Args:
+             resorce_client: Resource client that is being called
+        """
+        self._resource_client = resorce_client
+
+    def test_should_call_fail_json_when_oneview_exception(self):
+        self.mock_ansible_module.params = self.params_present
+        fake_exception = Mock(side_effect=HPOneViewException(self.error_message))
```

Comment:
Since self._resource_client is already a Mock, it is not necessary to instantiate another Mock, it could be simpler:
```python
fake_exception = HPOneViewException(self.error_message)
```